dockertest/images.py

```diff
@@ -54,8 +54,11 @@ class DockerImage(object):  # pylint: disable=R0902
     #: parsing, spec defined in docker-io documentation.  e.g.
     #: ``[registry_hostname[:port]/][user_name/]
     #: (repository_name[:version_tag])``
-    repo_split_p = re.compile(r"(.+?(:[\d]+?)?/)?([\w\-\.\+]+/)?"
-                              r"([\w\-\.]+)(:[\w\-\.]+)?")
+    repo_split_p = re.compile(r"(?P<repo_addr>.+?"  # optional
```

Comment:
If you add an extra layer of parens:

     r"((?P<repo_addr> [...] )?)/)?"  # new paren after r" and immediately before the slash


...you can remove the awkward trim-slash at line 145-146.